test/acceptance/features/steps/steps.py

```diff
@@ -145,26 +152,35 @@ def db_instance_is_running(context, db_name):
 @when(sbr_is_applied_step)
 def sbr_is_applied(context):
     sbr_yaml = context.text
-    sbr = ServiceBindingRequest()
-    if context.__contains__("nodejs_app"):
-        application = context.nodejs_app
-        context.nodejs_app_original_generation = application.get_observed_generation()
-        context.nodejs_app_original_pod_name = application.get_running_pod_name()
+    namespace = context.namespace
+    sbr = ServiceBindingRequest(namespace.name)
+    if context.__contains__("application"):
```

Comment:
how do we handle a case when this resoves to `false`? Can it happen at all?

Reply:
this is a condition where if application is involved then only captures the generation number and for all other yaml to be applied it skips tihs `if` condition and proceeds.